src/MagnumPlugins/BasisImporter/Test/BasisImporterTest.cpp

```diff
@@ -0,0 +1,188 @@
+    addInstancedTests({&BasisImporterTest::rgb,
+            &BasisImporterTest::rgba}, numTargetFormats);
+
+    /* Load the plugin directly from the build tree. Otherwise it's static and
+       already loaded. */
+    #ifdef BASISIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_manager.load(BASISIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+}
+
+void BasisImporterTest::empty() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+    char a{};
+    /* Explicitly checking non-null but empty view */
+    CORRADE_VERIFY(!importer->openData({&a, 0}));
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::openData(): the file is empty\n");
+}
+
+void BasisImporterTest::invalid() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!importer->openFile(Utility::Directory::join(PNGIMPORTER_TEST_DIR, "rgb.png")));
+
+    CORRADE_COMPARE(out.str(), Utility::formatString("Trade::BasisImporter::openData(): invalid header\n"));
+}
+
+void BasisImporterTest::openTwice() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    CORRADE_VERIFY(importer->openFile(
+        Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+    CORRADE_VERIFY(importer->openFile(
+        Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+
+    /* Shouldn't crash, leak or anything */
+}
+
+void BasisImporterTest::importTwice() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+
+    /* Verify that everything is working the same way on second use */
+    {
+        Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+        CORRADE_VERIFY(image);
+        CORRADE_COMPARE(image->size(), (Vector2i{3, 2}));
+    } {
+        Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+        CORRADE_VERIFY(image);
+        CORRADE_COMPARE(image->size(), (Vector2i{3, 2}));
+    }
+}
+
+void BasisImporterTest::configValue() {
+    typedef Corrade::Utility::ConfigurationValue<BasisTranscodingType> ConfigValue;
+    CORRADE_COMPARE("Etc1", ConfigValue::toString(BasisTranscodingType::Etc1, {}));
+
+    CORRADE_COMPARE("<invalid>", ConfigValue::toString(BasisTranscodingType(-1), {}));
+    CORRADE_COMPARE("<invalid>", ConfigValue::toString(BasisTranscodingType(42), {}));
+}
+
+void BasisImporterTest::rgb() {
+    const size_t instanceId = testCaseInstanceId();
+    const std::string suffix = (instanceId == 0) ? "" :
+        Corrade::Utility::ConfigurationValue<BasisTranscodingType>::toString(BasisTranscodingType(instanceId-1), {});
```

Comment:
Maybe use the canonical instance data definition instead? It'll make the code cleaner, free of the `?:` things and that way you can use the names directly, annotate the test cases etc.

```cpp
const struct {
    const char* format;
} FormatData[] {
    {"Etc2"},
    {"PvrtcBla"},
    ...
};

...

BasisImporterTest::BasisImporterTest() {
    addInstancedTests({...}, 
        Containers::arraySize(FormatData));
}

...

void BasisImporterTest::rgb() {
    auto& data = FormatData[testCaseInstanceId()];
    setTestCaseDescription(data.format);

    ...
```